host-configs/docker/gcc@12.3.0_cuda.cmake

```diff
@@ -0,0 +1,156 @@
+
+set(CMAKE_BUILD_TYPE "Release" CACHE STRING "")
+
+#------------------------------------------------------------------------------
+# Compilers
+#------------------------------------------------------------------------------
+# Compiler Spec: gcc@=12.3.0
+#------------------------------------------------------------------------------
+if(DEFINED ENV{SPACK_CC})
+
+  set(CMAKE_C_COMPILER "/home/serac/serac_tpls/spack/lib/spack/env/gcc/gcc" CACHE PATH "")
+
+  set(CMAKE_CXX_COMPILER "/home/serac/serac_tpls/spack/lib/spack/env/gcc/g++" CACHE PATH "")
+
+  set(CMAKE_Fortran_COMPILER "/home/serac/serac_tpls/spack/lib/spack/env/gcc/gfortran" CACHE PATH "")
+
+else()
+
+  set(CMAKE_C_COMPILER "/usr/bin/gcc" CACHE PATH "")
+
+  set(CMAKE_CXX_COMPILER "/usr/bin/g++" CACHE PATH "")
+
+  set(CMAKE_Fortran_COMPILER "/usr/bin/gfortran-12" CACHE PATH "")
+
+endif()
+
+set(CMAKE_C_FLAGS "-fPIC -pthread" CACHE STRING "")
+
+set(CMAKE_CXX_FLAGS "-fPIC -pthread" CACHE STRING "")
+
+#------------------------------------------------------------------------------
+# MPI
+#------------------------------------------------------------------------------
+
+set(MPI_C_COMPILER "/usr/bin/mpicc" CACHE PATH "")
+
+set(MPI_CXX_COMPILER "/usr/bin/mpic++" CACHE PATH "")
+
+set(MPI_Fortran_COMPILER "/usr/bin/mpif90" CACHE PATH "")
+
+set(MPIEXEC_EXECUTABLE "/usr/bin/mpirun" CACHE PATH "")
+
+set(MPIEXEC_NUMPROC_FLAG "-np" CACHE STRING "")
+
+set(ENABLE_MPI ON CACHE BOOL "")
+
+#------------------------------------------------------------------------------
+# Hardware
+#------------------------------------------------------------------------------
+
+#------------------------------------------------
+# Cuda
+#------------------------------------------------
+
+set(CUDAToolkit_ROOT "/usr/local/cuda-12.3" CACHE PATH "")
+
+set(CMAKE_CUDA_COMPILER "${CUDAToolkit_ROOT}/bin/nvcc" CACHE PATH "")
+
+set(CMAKE_CUDA_HOST_COMPILER "${CMAKE_CXX_COMPILER}" CACHE PATH "")
+
+set(CUDA_TOOLKIT_ROOT_DIR "/usr/local/cuda-12.3" CACHE PATH "")
+
+set(CMAKE_CUDA_ARCHITECTURES "70" CACHE STRING "")
+
+set(CMAKE_CUDA_FLAGS "" CACHE STRING "")
+
+set(ENABLE_OPENMP ON CACHE BOOL "")
+
+set(ENABLE_CUDA ON CACHE BOOL "")
+
+set(CMAKE_CUDA_SEPARABLE_COMPILATION ON CACHE BOOL "")
+
+set(CMAKE_CUDA_FLAGS "-fPIC -pthread${CMAKE_CUDA_FLAGS} --expt-extended-lambda --expt-relaxed-constexpr " CACHE STRING "" FORCE)
```

Comment:
@white238 I was wondering if the random `${CMAKE_CUDA_FLAGS}` here is right. You can see it here: https://github.com/LLNL/serac/blob/develop/scripts/spack/packages/serac/package.py#L301

Reply:
The point of that is to inherit anything set from the `CachedCMakePackage`. But there should be a space at the start of that line you linked.